velox/docs/functions/spark/datetime.rst

```diff
@@ -51,6 +51,39 @@ These functions support TIMESTAMP and DATE input types.
     ``num_days`` can be positive or negative.
     Supported types for ``num_days`` are: TINYINT, SMALLINT, INTEGER.
 
+.. spark:function:: date_trunc(fmt, ts) -> timestamp
+
+    ``fmt`` is case insensitive and must be one of the following:
+        * "YEAR", "YYYY", "YY" - truncate to the first date of the year that the ``ts`` falls in, the time part will be zero out
+        * "QUARTER" - truncate to the first date of the quarter that the ``ts`` falls in, the time part will be zero out
+        * "MONTH", "MM", "MON" - truncate to the first date of the month that the ``ts`` falls in, the time part will be zero out
+        * "WEEK" - truncate to the Monday of the week that the ``ts`` falls in, the time part will be zero out
+        * "DAY", "DD" - zero out the time part
+        * "HOUR" - zero out the minute and second with fraction part
+        * "MINUTE"- zero out the second with fraction part
+        * "SECOND" - zero out the second fraction part
+        * "MILLISECOND" - zero out the microseconds
+        * "MICROSECOND" - everything remains
+
+    Returns timestamp ``ts`` truncated to the unit specified by the format model ``fmt``.
+    Returns null if ``fmt`` is invalid. ::
```

Comment:
What if `ts` is null?

Reply:
return null.

Reply:
https://github.com/facebookincubator/velox/pull/11340/files#diff-f7042b580cdd92f6b4ba8b77183933cbb460c88a6deed81f767a35bb4d2cc40eR1112